pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
Yes.  It's a little confusing because of my workaround for the bug in the existing/old libkv+etcd.  I have to set up both the ClientTLS as well as TLS, but can confirm that the TLS field is used, which I've wired up as @NathanMcCauley requested.